FitGoal/FitGoal/HomeViewController/CollectionViewSectionsAndHeaders/RoutineInspectorCell.swift

```diff
@@ -52,6 +56,12 @@ class RoutineInspectorCell: UICollectionViewCell {
         routineExercises = exercises
         routineCollectionView.reloadData()
     }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard let cell = routineCollectionView.cellForItem(at: indexPath) as? ExerciseCell else { return }
+        guard let exercise = cell.exercise else { return }
+        routineInspectorCellDelegate?.userDidSelectExersice(exercise)
+    }
```

Comment:
Instead of getting the exercise from the cell, you should get the exercise from the list of exercises you have:

```swift
let exercise = routineExercises[indexPath.item]
```